### PR TITLE
Add log lookup exception for empty op subtypes

### LINKED ISSUE #34228: DAGs log view of `EmptyOperator` shows confusing error `Request URL missing protocol`.
### Apache Airflow version

2.7.1

### What happened

At airflow log view (DAG -> task -> log) of `EmptyOperator`, an error shows in window:

```
*** Could not read served logs: Request URL is missing an 'http://' or 'https://' protocol.
```

Full trace in console:

```
 webserver | [2023-09-09T09:26:47.632+0800] {file_task_handler.py:524} ERROR - Could not read served logs
 webserver | Traceback (most recent call last):
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_transports/default.py", line 60, in map_httpcore_exceptions
 webserver | yield
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_transports/default.py", line 218, in handle_request
 webserver | resp = self._pool.handle_request(req)
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpcore/_sync/connection_pool.py", line 214, in handle_request
 webserver | raise UnsupportedProtocol(
 webserver | httpcore.UnsupportedProtocol: Request URL is missing an 'http://' or 'https://' protocol.
 webserver |
 webserver | The above exception was the direct cause of the following exception:
 webserver |
 webserver | Traceback (most recent call last):
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/airflow/utils/log/file_task_handler.py", line 507, in _read_from_logs_server
 webserver | response = _fetch_logs_from_service(url, rel_path)
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/airflow/utils/log/file_task_handler.py", line 90, in _fetch_logs_from_service
 webserver | response = httpx.get(
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_api.py", line 189, in get
 webserver | return request(
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_api.py", line 100, in request
 webserver | return client.request(
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_client.py", line 814, in request
 webserver | return self.send(request, auth=auth, follow_redirects=follow_redirects)
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_client.py", line 901, in send
 webserver | response = self._send_handling_auth(
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_client.py", line 929, in _send_handling_auth
 webserver | response = self._send_handling_redirects(
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_client.py", line 966, in _send_handling_redirects
 webserver | response = self._send_single_request(request)
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_client.py", line 1002, in _send_single_request
 webserver | response = transport.handle_request(request)
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_transports/default.py", line 217, in handle_request
 webserver | with map_httpcore_exceptions():
 webserver | File "/nix/store/bc45k1n0pkrdkr3xa6w84w1xhkl1kkyp-python3-3.10.12/lib/python3.10/contextlib.py", line 153, in __exit__
 webserver | self.gen.throw(typ, value, traceback)
 webserver | File "/persistent/Works/apache-airflow/venv/lib/python3.10/site-packages/httpx/_transports/default.py", line 77, in map_httpcore_exceptions
 webserver | raise mapped_exc(message) from exc
 webserver | httpx.UnsupportedProtocol: Request URL is missing an 'http://' or 'https://' protocol.
```

### What you think should happen instead

Old version of airflow (I tested 2.6.0), log view of `EmptyOperator` will show noting,  
I'm not sure this behavior is documented, but It caused confusion for me at least.

### How to reproduce

1. I use python venv with whl in GitHub releases for installation.
2. Run `airflow standalone` for development deploy, I also disable load_examples.
3. Using `EmptyOperator` example like below, I also disable catchup of DAG:

```Python
import datetime

from airflow import DAG
from airflow.operators.empty import EmptyOperator

with DAG(
    dag_id="demo",
    start_date=datetime.datetime(2021, 1, 1),
    schedule="@daily",
    catchup=False,
):
    EmptyOperator(task_id="demo")
```

4. Enable the DAG in airflow webserver, wait it for a run.
5. Navigate to demo dag, demo task; press log view at toolbar.

### Operating System

NixOS 23.05 (Stoat)

### Versions of Apache Airflow Providers

```
apache-airflow-providers-common-sql==1.7.1
apache-airflow-providers-ftp==3.5.1
apache-airflow-providers-http==4.5.1
apache-airflow-providers-imap==3.3.1
apache-airflow-providers-sqlite==3.4.3
```

### Deployment

Virtualenv installation

### Deployment details

Our enterprise internal deployment using modified docker-compose, version 2.7.0 shows the same issue.
But our old 2.5.2 venv deployment works fine. `EmptyOperator` shows nothing as normal.

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by vchiapaikeo**
closes: https://github.com/apache/airflow/issues/34228

image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by eladkal** [comment]
@vchiapaikeo are you still working on this PR?

**Comment by vchiapaikeo** [comment]
Whoops, sorry about that. Rebased and ready for review here. I retested and it still works as expected -

image

File: airflow/utils/log/file_task_handler.py [hunk 1]
```diff
@@ -357,7 +367,9 @@ def _read(
             worker_log_full_path = Path(self.local_base, worker_log_rel_path)
             local_messages, local_logs = self._read_from_local(worker_log_full_path)
             messages_list.extend(local_messages)
-        if is_running and not executor_messages:
+        if ti.task.inherits_from_empty_operator is True and self._test_mode is False:
```
Comment (eladkal) [inline airflow/utils/log/file_task_handler.py]: Why does it matter if we are in test mode or not?

Comment (vchiapaikeo) [reply airflow/utils/log/file_task_handler.py]: So this change actually causes several tests to fail because many tests check to see that the following call (`served_messages, served_logs = self._read_from_logs_server(ti, worker_log_rel_path`)) is made. Let me take another look at this and see if we can do the override for EmptyOperator while still keeping these tests intact.

Comment (vchiapaikeo) [reply airflow/utils/log/file_task_handler.py]: Okay, modified my approach here to catch the exception instead which is much less disruptive to tests. Retested manually as well and all continues to look good - 

Screenshot 2024-01-11 at 9 44 50 AM

File: airflow/utils/log/file_task_handler.py [hunk 2]
```diff
@@ -78,8 +79,6 @@ def _set_task_deferred_context_var():
 
 
 def _fetch_logs_from_service(url, log_relative_path):
-    import httpx
-
```
Comment (eladkal) [inline airflow/utils/log/file_task_handler.py:82]: Why was this import moved to top of the file? is it due to pre commit change?

Comment (vchiapaikeo) [reply airflow/utils/log/file_task_handler.py:82]: No, I did this because I need this module on line 506 (httpx.UnsupportedProtocol). I looked at the blame and there didn't seem to be any reason for this to be function scoped so moving it to module scoped.

**Review by eladkal — APPROVED** [review]

File: airflow/utils/log/file_task_handler.py (hunk 2 above)
Comment (vchiapaikeo) [reply airflow/utils/log/file_task_handler.py:82]: Actually, it looks like Ash moves this out of module scope to function for perf optimization here: https://github.com/apache/airflow/commit/1a8a897120762692ca98ac5ce4da881678c073aa#diff-e7f34f73940eb52d92bb991abedc1c963431c5373c12dff739c8fb7d03e93d3aL24

Going to move it back to function scope in a follow up PR

Comment (vchiapaikeo) [reply airflow/utils/log/file_task_handler.py:82]: PR: https://github.com/apache/airflow/pull/36753

**Comment by zachliu** [comment]
I believe this bug fix has been invalidated by https://github.com/apache/airflow/pull/39799
instead of https://github.com/apache/airflow/blob/40cb51d135094c6eca7c69e75fb192f9d1113bde/airflow-core/src/airflow/utils/log/file_task_handler.py#L608
it should be
```python
from requests.exceptions import InvalidURL
```
